pool/pool.go

```diff
@@ -30,284 +31,576 @@ import (
+	ContainerSetEACL(context.Context, sdkClient.PrmContainerSetEACL) (*sdkClient.ResContainerSetEACL, error)
+	EndpointInfo(context.Context, sdkClient.PrmEndpointInfo) (*sdkClient.ResEndpointInfo, error)
+	NetworkInfo(context.Context, sdkClient.PrmNetworkInfo) (*sdkClient.ResNetworkInfo, error)
+	ObjectPutInit(context.Context, sdkClient.PrmObjectPutInit) (*sdkClient.ObjectWriter, error)
+	ObjectDelete(context.Context, sdkClient.PrmObjectDelete) (*sdkClient.ResObjectDelete, error)
+	ObjectGetInit(context.Context, sdkClient.PrmObjectGet) (*sdkClient.ObjectReader, error)
+	ObjectHead(context.Context, sdkClient.PrmObjectHead) (*sdkClient.ResObjectHead, error)
+	ObjectRangeInit(context.Context, sdkClient.PrmObjectRange) (*sdkClient.ObjectRangeReader, error)
+	ObjectSearchInit(context.Context, sdkClient.PrmObjectSearch) (*sdkClient.ObjectListReader, error)
+	SessionCreate(context.Context, sdkClient.PrmSessionCreate) (*sdkClient.ResSessionCreate, error)
+}
+
+// InitParameters contains values used to initialize connection Pool.
+type InitParameters struct {
+	key                       *ecdsa.PrivateKey
+	logger                    *zap.Logger
+	nodeConnectionTimeout     time.Duration
+	nodeRequestTimeout        time.Duration
+	clientRebalanceInterval   time.Duration
+	sessionTokenThreshold     time.Duration
+	sessionExpirationDuration uint64
+	nodeParams                []NodeParam
+
+	clientBuilder func(endpoint string) (client, error)
+}
+
+// SetKey specifies default key to be used for the protocol communication by default.
+func (x *InitParameters) SetKey(key *ecdsa.PrivateKey) {
+	x.key = key
+}
+
+// SetLogger specifies logger.
+func (x *InitParameters) SetLogger(logger *zap.Logger) {
+	x.logger = logger
+}
+
+// SetNodeConnectionTimeout specifies the timeout for connection to be established.
+func (x *InitParameters) SetNodeConnectionTimeout(timeout time.Duration) {
+	x.nodeConnectionTimeout = timeout
+}
+
+// SetNodeRequestTimeout specifies the timeout for request to node to decide if it is alive.
+func (x *InitParameters) SetNodeRequestTimeout(timeout time.Duration) {
+	x.nodeRequestTimeout = timeout
+}
+
+// SetClientRebalanceInterval specifies the interval for updating nodes health status.
+func (x *InitParameters) SetClientRebalanceInterval(interval time.Duration) {
+	x.clientRebalanceInterval = interval
+}
+
+// SetSessionTokenThreshold specifies the max session token life time for PutObject operation.
+func (x *InitParameters) SetSessionTokenThreshold(threshold time.Duration) {
+	x.sessionTokenThreshold = threshold
+}
+
+// SetSessionExpirationDuration specifies the session token lifetime in epochs.
+func (x *InitParameters) SetSessionExpirationDuration(expirationDuration uint64) {
+	x.sessionExpirationDuration = expirationDuration
+}
+
+// SetNodeParams specifies information about the nodes to which you want to connect.
+func (x *InitParameters) SetNodeParams(nodeParams []NodeParam) {
```

Comment:
Don't u think that 

```suggestion
func (x *InitParameters) AddNode(nodeParams NodeParam) {
```

would be more convenient?